pxr/usd/usdShade/testenv/testUsdShadeValidators.cpp

```diff
@@ -557,6 +559,213 @@ TestUsdShadeEncapsulationRulesValidator()
     }
 }
 
+void ValidateError(const UsdValidationErrorVector& errors,
+        const TfToken& expectedErrorIdentifier,
+        const SdfPath& expectedPrimPath,
+        const std::string& expectedErrorMsg,
+        UsdValidationErrorType expectedErrorType = UsdValidationErrorType::Error)
+{
+    TF_AXIOM(errors.size() == 1);
+    TF_AXIOM(errors[0].GetIdentifier() == expectedErrorIdentifier);
+    TF_AXIOM(errors[0].GetType() == expectedErrorType);
+    TF_AXIOM(errors[0].GetSites().size() == 1);
+    TF_AXIOM(errors[0].GetSites()[0].IsValid());
+    TF_AXIOM(errors[0].GetSites()[0].IsPrim());
+    TF_AXIOM(errors[0].GetSites()[0].GetPrim().GetPath() ==
+             expectedPrimPath);
+    TF_AXIOM(errors[0].GetMessage() == expectedErrorMsg);
+}
+
+void
+TestUsdShadeNormalMapTextureValidator()
```

Comment:
@tallytalwar There are quite a few failure conditions in this test and validator, let me know if you think it would be better to break this up in to multiple validators or not